kedro/contrib/io/matplotlib/matplotlib_writer.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2018-2019 QuantumBlack Visual Analytics Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+``MatplotlibWriter `` saves matplotlib objects as image files.
+"""
+
+from pathlib import Path
+from typing import Any, Dict
+
+from matplotlib.pyplot import figure
+
+from kedro.io import AbstractDataSet, DataSetError
+
+
+class MatplotlibWriter(AbstractDataSet):
+    """``MatplotlibWriter`` saves matplotlib objects as image files.
+
+        Example:
+        ::
+
+            >>> import matplotlib.pyplot as plt
+            >>> from kedro.contrib.io.matplotlib import MatplotlibWriter
+            >>>
+            >>> plt.plot([1,2,3],[4,5,6])
+            >>>
+            >>> single_plot_writer = MatplotlibWriter(filepath="data/new_plot.png")
+            >>> single_plot_writer.save(plt)
+            >>>
+            >>> plt.close()
+            >>>
+            >>> plots = dict()
+            >>>
+            >>> for colour in ['blue', 'green', 'red']:
+            >>>     plots[colour] = plt.figure()
+            >>>     plt.plot([1,2,3],[4,5,6], color=colour)
+            >>>     plt.close()
+            >>>
+            >>> multi_plot_writer = MatplotlibWriter(filepath="data/")
+            >>> multi_plot_writer.save(plots)
+
+    """
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(
+            filepath=self._filepath,
+            load_args=self._load_args,
+            save_args=self._save_args,
+        )
```

Comment:
Small thing, but I'd rather have `__init__` the first function the user sees when they dig out the source code.